internal/services/containers/kubernetes_addons.go

```diff
@@ -235,6 +239,50 @@ func schemaKubernetesAddOnProfiles() *pluginsdk.Schema {
 						},
 					},
 				},
+				"azure_keyvault_secrets_provider": {
+					Type:     pluginsdk.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &pluginsdk.Resource{
+						Schema: map[string]*pluginsdk.Schema{
+							"enabled": {
+								Type:     pluginsdk.TypeBool,
+								Required: true,
+							},
+							"secret_rotation_enabled": {
+								Type:     pluginsdk.TypeBool,
+								Default:  false,
+								Optional: true,
+							},
+							"secret_rotation_interval": {
+								Type:         pluginsdk.TypeString,
+								Optional:     true,
+								Default:      "2m",
+								ValidateFunc: containerValidate.Duration,
+							},
+							"azure_keyvault_secrets_provider_identity": {
```

Comment:
Sorry I missed this one in the initial review, since this block is already called `azure_keyvault_secrets_provider` could we rename this to just `identity`

Reply:
I think the  convention might be there to prevent confusion?

Looking at the AKS module docs, it might get a bit confusing if there were multiple `The identity block exports the following:` sections, which I guess is why each identity has its own name?

Reply:
If the schemas for each identity block were different I would agree, but they're all identical so I don't really see an issue if they all point to the same block in the docs. To be sure I will clarify internally whether there is any other reason for that naming convention. If not it should be renamed and the other two properties will also need to be renamed and deprecated (in a separate PR).

Reply:
Could also be that the AKS cluster itself already exports an `identity` block, which does have a different schema from the addon identity blocks?

Reply:
I think it would add confusion with the AKS `identity` and `kubelet_identity` blocks like you said. Maybe something more generic for the addons (in a future PR) like `addon_identity`? It's always going to be the same with a `user_assigned_managed_identity` anyways.